langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal

```diff
@@ -517,6 +517,68 @@ type Person2 record {
     int age;
 };
 
+function testCloneWithTypeTupleToJSON() {
+    [string, string, string] tupleValue1 = ["Mohan", "single", "LK2014"];
+    json|error jsonValue = tupleValue1.cloneWithType();
+
+    assert(jsonValue is error, false);
+    assert(jsonValue is json[], true);
+
+    [string, string, xml] tupleValue2 = ["Mohan", "single"];
+    jsonValue = tupleValue2.cloneWithType();
+    assert(jsonValue is error, true);
+     if (jsonValue is error) {
```

Comment:
Instead of using a type guard, we can safely cast here instead. 

```ballerina
error a = <error> jsonValue;
```

That way we can get rid of the indented blocks. Applies in other places too.